app/controllers/memberships_controller.rb

```diff
@@ -38,8 +38,9 @@ def accept
   def withdraw
     Membership::WithdrawService.call(membership)
 
-    redirect_back(fallback_location: group_path(group))
-  rescue Membership::WithdrawService::MembershipMustHaveDefaultPost
+    redirect_to(group_path(group)) if membership.user == current_user
+  rescue Membership::WithdrawService::MembershipMustHaveDefaultPost,
+         Membership::WithdrawService::MembershipHasPointRequests
```

Comment:
What about using a common parent class here?
```suggestion
  rescue Membership::WithdrawService::WithdrawError
```